airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/ctid/PostgresCtidHandler.java

```diff
@@ -174,8 +182,15 @@ private AutoCloseableIterator<AirbyteMessage> augmentWithLogs(final AutoCloseabl
 
   private AutoCloseableIterator<AirbyteMessage> augmentWithState(final AutoCloseableIterator<AirbyteMessage> recordIterator,
       final AirbyteStreamNameNamespacePair pair) {
+
+    final CtidStatus currentCtidStatus = ctidStateManager.getCtidStatus(pair);
+    final JsonNode incrementalState =
+        (currentCtidStatus == null || currentCtidStatus.getIncrementalState() == null) ? streamStateForIncrementalRunSupplier.apply(pair)
+            : currentCtidStatus.getIncrementalState();
```

Comment:
What is `currentCtidStatus.getIncrementalState()` returning on a new stream?
If `null`, is ling 193 going to accept a null?

Reply:
Thats what the null check is for. If its null then we get the incremental state via `streamStateForIncrementalRunSupplier.apply(pair)`